clang/lib/Serialization/ASTReaderDecl.cpp

```diff
@@ -2674,7 +2674,7 @@ void ASTDeclReader::VisitTemplateTypeParmDecl(TemplateTypeParmDecl *D) {
 
   D->setDeclaredWithTypename(Record.readInt());
 
-  if (D->hasTypeConstraint()) {
+  if (Record.readBool() && D->hasTypeConstraint()) {
```

Comment:
Can you put that bool in a variable? It would be more clear what it is

Reply:
Sure. Just waiting on a final verdict, on whether this is still needed.